torchmetrics/retrieval/retrieval_metric.py

```diff
@@ -82,18 +82,18 @@ def __init__(
 
         empty_target_action_options = ('error', 'skip', 'neg', 'pos')
         if empty_target_action not in empty_target_action_options:
-            raise ValueError(f"`empty_target_action` received a wrong value `{empty_target_action}`.")
+            raise ValueError(f"Argument `empty_target_action` received a wrong value `{empty_target_action}`.")
 
         self.empty_target_action = empty_target_action
 
         self.add_state("indexes", default=[], dist_reduce_fx=None)
         self.add_state("preds", default=[], dist_reduce_fx=None)
         self.add_state("target", default=[], dist_reduce_fx=None)
 
-    def update(self, preds: Tensor, target: Tensor, indexes: Tensor = None) -> None:
+    def update(self, preds: Tensor, target: Tensor, indexes: Tensor) -> None:  # type: ignore
```

Comment:
@maximsch2 any suggestion how to deal with this 
```
error: Signature of "update" incompatible with supertype "Metric"  [override]
```

Reply:
@Borda I approved since everything else looks fine but let's not have this PR merged before this is resolved (this is probably the main issue of metrics typing right now :) )

Reply:
lets merge it and address this specific case in separate PR

Reply:
@justusschock @SkafteNicki @Borda : Since the metric base class uses *args, **kwargs this is currently unavaoidable. Is there a suppression we can add in this case? 

Alternatively, why we need to specify *args, **kwargs in the metric base class? This is entirely controlled by the concrete implementation

Reply:
it is because of the internal wrapper, see https://github.com/PyTorchLightning/metrics/blob/eaf85cf42b6e58cf3c631e214154e33414fd75fb/torchmetrics/metric.py#L180